### PR TITLE
Add bindF

### PR DESCRIPTION
**Pull request by garyb**
There's perhaps a better/more correct definition for this comment, recommendations would be appreciated. If everyone's happy I'll merge this one before the `injF` / `injFC` change as it can be done as a non-breaking change first.

### CODE REVIEW

**Comment by paf31** [comment]
:+1: Looks good to me!

An example of this might be neat too, later. Maybe you could have `data Echo a = Echo a` interpreted in terms of `Teletype` or something, by using two commands to echo the user's input.

File: src/Control/Monad/Free.purs [hunk 1]
```diff
@@ -94,6 +95,10 @@ pureFC = liftFC <<< pure
 mapF :: forall f g a. (Functor f, Functor g) => Natural f g -> Free f a -> Free g a
 mapF t fa = either (\s -> Free <<< t $ mapF t <$> s) Pure (resume fa)
 
+-- | Use a natural transformation to interpret one `Free` monad as another.
+bindF :: forall f g a. (Functor f, Functor g) => Natural f (Free g) -> Free f a -> Free g a
```
Comment (jdegoes) [inline src/Control/Monad/Free.purs]: Just one small comment, the order of parameters does not match `bind`. Otherwise, :+1:

**Comment by garyb** [comment]
Good points both, I've reordered the arguments to match `bind` and added an example. The `Echo` example would perhaps have been better, but I didn't read this properly so just made up my own case. :smile:

**Comment by jdegoes** [comment]
Awesome. :+1: 

At some point, I'd love higher-kinded type classes for Functor, Monad, etc. :smiling_imp:

**Pull request merged**